### ISSUE TITLE
Invalid pointer when running aggregation 

### ISSUE THREAD

**Issue by Jap8nted** [holoviz/datashader#1141]
I am running fairly straight forward process of passing a Dask DataFrame containing many points (lat, lon) and other dimensions to render to a canvas. Afterwards I run mean of one of those dimensions I get the error: 
```munmap_chunk(): invalid pointer
qemu: uncaught target signal 6 (Aborted) - core dumped
```
Is there anything I can check to find out the reason for this to generate an invalid pointer? it is clearly the data inside which generates this because with other datasets I get no error. 

#### ALL software version info
python 3.9.7

numba==0.56.3
pandas==1.3.5
dask==2022.5.0
numpy==1.22.0
datashader==0.14.2

```
from datashader import Canvas
from datashader.reductions import mean, sum
ddf = dask.read_csv("file.csv")
cvs = Canvas(plot_width=50, plot_height=17)
cvs.points(ddf, "LON", "LAT", agg=mean("VALUE"))
```

<img width="887" alt="image" src="https://user-images.githubusercontent.com/5432705/196892834-014e36ad-c16f-4ecd-b9e0-4ea0c9266ea9.png">

**Comment by jbednar** [comment]
It would be very difficult for us to reproduce a problem like that from that description. I realize that it may be difficult for you to provide a reproducible example for a big run like that, but could you answer some questions?

- Are you running this on an M1 Mac? If so it might be an issue with the Intel processor implementation unrelated to Datashader (see e.g. https://github.com/docker/for-mac/issues/5148#issuecomment-784992663). You could try running it in native environment instead, which should also be faster.
- Are you saying that you get the error only on some datasets, but when you do get it, it's always on the `cvs.points` call? If so that warning might indicate that the entire aggregate is zero, which I'm pretty sure we have tests for, but perhaps not for this specific case using a mean aggregator. If so, maybe you could make a simple test case that would reproduce the issue.

**Comment by Jap8nted** [comment]
Hi,

Sorry for the late response, I was long time away. The bug actually pops up when a dask dataframe is passed to cvs.points aggregating when the canvas has either width=0 or height=0,  I updated the versions of datashader and dask and run on an M1 Mac and a a Ubuntu amd64, the problem appeared on the amd64. Previously we got the core dump on a pod running on an EKS cluster. 


```
from datashader import Canvas
from datashader.reductions import mean, sum, max, min
from dask import dataframe as ddf
def main():
    df = pd.DataFrame({"LON": [0, 1, 2, 3, 4], "LAT": [0, 1, 2, 3, 4], "VALUE": [0, 1, 2, 3, 4]})
    data = ddf.from_pandas(df, npartitions=1)
    cvs = Canvas(
        plot_width=0,
        plot_height=0
    )
    cvs.points(data, "LON", "LAT", agg=max("VALUE"))
if __name__ == "__main__":
    main()
```

I thought on just adding validation as shown on the PR: https://github.com/holoviz/datashader/pull/1183 but am not sure what is the best way to tacke this. I debugged the code and the error appears when Dask graph is scheduled on line https://github.com/holoviz/datashader/blob/0ade086bd3056d740a6ff42a2f79d13275e6c273/datashader/data_libraries/dask.py#L29

So I am not sure is the right way of tackling this is preventing here o investigating the reason on Dask

**Comment by philippjfr** [comment]
Sounds like we should just skip aggregation entirely if width/height is None since no actual computation is required to return an empty array.

**Comment by jbednar** [comment]
Yes, if either dimension is zero width, return without aggregating.

**Comment by ianthomas23** [comment]
Assuming we'd still want to return a valid `xr.DataArray` or `xr.Dataset` with correct `dims`, etc, then I think the lines to avoid calling for `pandas` if `width*height == 0` would be 
https://github.com/holoviz/datashader/blob/198c0b46a00a9dafe2a98a90aa4327864c4938d0/datashader/data_libraries/pandas.py#L31
and
https://github.com/holoviz/datashader/blob/198c0b46a00a9dafe2a98a90aa4327864c4938d0/datashader/data_libraries/pandas.py#L48
Similar would be required for `dask`, `cudf`, etc.

**Comment by Jap8nted** [comment]
For pandas it works without problem, an empty array is getting returned. For cudf, have not tried though, so I will implement the dask, cudf to achieve consistent behavior.

**Comment by ianthomas23** [comment]
Thanks @Jap8nted, that will be great.

**Comment by ianthomas23** [comment]
Closed by #1183.
